docs/src/main/tut/datatypes/validated.md

```diff
@@ -12,7 +12,310 @@ Response comes back saying your username can't have dashes in it, so you make so
+    }
+      yield RegistrationData(validatedUserName, validatedPassword, validatedFirstName, validatedLastName, validatedAge)
+  }
+
+}
+
+object FormValidatorValidated extends FormValidatorValidated
+```
+
+Looks similar to the first version. What we've done here was to use `Validated` instead of `Either`. Please note that our `Right` is now a `Valid` and `Left` is an `Invalid`.
+Remember, our goal is to get all the validation errors for displaying it to the user.
+
+But this approach won't compile. Why? Here is an excerpt of the compiler output:
+
+```
+value flatMap is not a member of cats.data.Validated[DomainValidation,String]
+```
+
+The problem lies here:
+
+```tut:silent:fail
+ for {
+      validatedUserName <- validateUserName(username)
+      validatedPassword <- validatePassword(password)
+      validatedFirstName <- validateFirstName(firstName)
+      validatedLastName <- validateLastName(lastName)
+      validatedAge <- validateAge(age)
+    }
+      yield RegistrationData(validatedUserName, validatedPassword, validatedFirstName, validatedLastName, validatedAge)
+```
+
+Without diving into details about monads, a for-comprehension uses the `flatMap` method for composition. Monads like `Either` can be composed in that way, but the thing with `Validated` is that it isn't a monad, but an [_Applicative Functor_](../typeclasses/applicativetraverse.html).
+
+So, how do we do here?
+
+### Meeting applicative
+
+We have to look into another direction: a for-comprehension plays well in a fail-fast scenario, but the structure in our previous example was designed to catch one error at a time, so, our next step is to tweak the implementation a bit.
+
+```tut:silent
+import cats.data._
+import cats.data.Validated._
+import cats.implicits._
+
+sealed trait FormValidatorNel {
+
+  type ValidationResult[A] = Validated[NonEmptyList[DomainValidation], A]
+
+  private def validateUserName(userName: String): ValidationResult[String] =
+    if (userName.matches("^[a-zA-Z0-9]+$")) userName.validNel else UsernameHasSpecialCharacters.invalidNel
+
+  private def validatePassword(password: String): ValidationResult[String] =
+    if (password.matches("(?=^.{10,}$)((?=.*\\d)|(?=.*\\W+))(?![.\\n])(?=.*[A-Z])(?=.*[a-z]).*$")) password.validNel
+    else PasswordDoesNotMeetCriteria.invalidNel
+
+  private def validateFirstName(firstName: String): ValidationResult[String] =
+    if (firstName.matches("^[a-zA-Z]+$")) firstName.validNel else FirstNameHasSpecialCharacters.invalidNel
+
+  private def validateLastName(lastName: String): ValidationResult[String] =
+    if (lastName.matches("^[a-zA-Z]+$")) lastName.validNel else LastNameHasSpecialCharacters.invalidNel
+
+  private def validateAge(age: Int): ValidationResult[Int] =
+    if (age >= 18 && age <= 75) age.validNel else AgeIsInvalid.invalidNel
+
+  def validateForm(username: String, password: String, firstName: String, lastName: String, age: Int): ValidationResult[RegistrationData] = {
```

Comment:
Can you perhaps put all the `validateN` functions into another snippet that's checked by tut, and then only have the `validateForm` go without `tut`? I don't think you need the surrounding trait btw :)

Reply:
Meant to put this on the snippet above 😅

Reply:
Got it! Working on it and also on the support of <2.11.x `Either` for-comprehensions (Travis complained about `flatMap` on this versions).

Thank you for your patience with this! :)

Reply:
Just import `cats.syntax.either._` to get that working :)

Thank you, for you help with the docs!